### ISSUE TITLE
make test failed in case of usign fips

### ISSUE THREAD

**Issue by juhaszviktor** [openssl/openssl#786]
We have compilation problem, when we tried to build the openssl 1.0.1s

test SSL protocol
test ssl3 is forbidden in FIPS mode
*** IN FIPS MODE ***
Available compression methods:
  1: zlib compression
47155267252048:error:140A9129:SSL routines:SSL_CTX_new:only tls allowed in fips mode:ssl_lib.c:1720:
47155267252048:error:140A9129:SSL routines:SSL_CTX_new:only tls allowed in fips mode:ssl_lib.c:1720:
test ssl2 is forbidden in FIPS mode
Testing was requested for a disabled protocol. Skipping tests.
make[1]: *** [test_ssl] Error 1
make[1]: Leaving directory `/usr/src/openssl/openssl-1.0.1s-1.syslogng50.12/test'
make: *** [tests] Error 2

**Comment by juhaszviktor** [comment]
I think this is caused by disabling ssl2 by default. But in this case the testfipsssl will fail,
because the ssl2 test will return 1

If I comment out the ```$ssltest -ssl2 $extra && exit 1```, then the make test will passed
```
echo test ssl2 is forbidden in FIPS mode
$ssltest -ssl2 $extra && exit 1
```

**Comment by juhaszviktor** [comment]
OK, problem is solved by patch: f4bed7c7b6fe8459dbc14e28f91c6150013b9c5e
